functors_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,29 @@
 #include "impl.h"
 
+BeerBrand BeerOrganizer::operator()() {
+    if (static_cast<size_t>(BeerBrand::Max) == currentIndex)
+        currentIndex = 1;
+
+    return static_cast<BeerBrand>(currentIndex++);
+}
+
+bool isExpensiveBeer(const BeerBrand& brand)
+{
+    std::vector<BeerBrand> expensive{BeerBrand::HoeGaarden, BeerBrand::Corona};
+
+    return std::find(expensive.begin(), expensive.end(), brand) != std::end(expensive);
+}
+
+Cocktail MixingPolicy::mix(AlcoholDrink alcohol, NonAlcoholDrink nonAlcohol)
+{
+    using Recipe = std::tuple<AlcoholDrink, NonAlcoholDrink>;
```

Comment:
why tuple and not the pair?

Reply:
Tuple was removed, now only pair is used.